Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -933,6 +1024,114 @@ void UpdatePlanes(std::vector<Frame::Plane::Identifier>& updatedPlanes, std::vec
             }
         }
 
+        /**
+         Updates existing meshes in place, gets the list of updated/created mesh IDs and removed mesh IDs.
+         */
+        void UpdateMeshes(std::vector<Frame::Mesh::Identifier>& updatedMeshes, std::vector<Frame::Mesh::Identifier>& deletedMeshes) {
+            if (!meshDetectionEnabled)
+            {
+                return;
+            }
+            
+            [sessionDelegate LockMeshes];
+            @try {
+                if (@available(iOS 13.4, *)) {
+                    // Update all meshes that have been updated since the last frame
+                    auto updatedARKitMeshes = [sessionDelegate GetUpdatedMeshes];
+                    for (ARMeshAnchor* updatedMesh : *updatedARKitMeshes) {
+                        auto geometry = updatedMesh.geometry;
+                        auto faces = geometry.faces;
+                        auto vertices = geometry.vertices;
+                        auto normals = geometry.normals;
```

Comment:
it seems like we may want to access these vertices, normals, faces by reference. I think right now you have  copy constructor declared. If we do something like

const auto& vertices = geometry.vertices;

or

auto& vertices = geometry.vertices;

i think you should be able to reference this data in its current location. This will be more performant than having to regenerate this data in the new vertices/normals/faces vectors you are creating.

Reply:
that all going said, im not 100% what data format these different values are in wrapped by the geometry. if this is only a reference copy to the data object it should be fine.

Reply:
Because faces, vertices, and normals are in MTL Buffers, I was having issues with accessing the data and converting it from bytes to the formats that I need (Vector3f & uint32_t) correctly. The numbers were always off. But I will look into this some more